proposals/api/views.py

```diff
@@ -0,0 +1,190 @@
+from braces.views import LoginRequiredMixin
+from django.db.models import Q
+from django.utils.translation import ugettext_lazy as _
+from rest_framework.authentication import SessionAuthentication
+
+from reviews.mixins import CommitteeMixin
+from uil.vue.rest import FancyListApiView
+
+from main.utils import is_secretary
+from .serializers import ProposalSerializer
+from ..models import Proposal
+
+
+class BaseProposalsApiView(LoginRequiredMixin, FancyListApiView):
```

Comment:
If I understand correctly, Django list views are now simply TemplateViews with a distinct data URL pointing to the API view.

Vue requests the data_url and then in the ApiView a serializer takes the queryset and squirts it out as JSON for Vue to display in the template. Seems pretty neat.

Reply:
100% correct :)